src/components/pill-selector-dropdown/PillSelectorDropdown.js

```diff
@@ -57,44 +59,46 @@ type Props = {
     /** Validate the given input value, and update `error` prop if necessary */
     validateForError?: Function,
     /** Called to check if pill item data is valid. The `item` is passed in. */
-    validator?: Function,
+    validator: (option: Option | OptionValue) => boolean,
 };
 
 type State = {
     inputValue: string,
+    isInCompositionMode: boolean,
 };
 
 class PillSelectorDropdown extends React.Component<Props, State> {
     static defaultProps = {
         allowCustomPills: false,
+        allowInvalidPills: false,
         disabled: false,
         error: '',
         inputProps: {},
         label: '',
         placeholder: '',
         selectedOptions: [],
         selectorOptions: [],
+        validator: () => true,
     };
 
-    state = { inputValue: '' };
+    state = { inputValue: '', isInCompositionMode: false };
 
     parsePills = () => {
         const { inputValue } = this.state;
-        const { parseItems, validator } = this.props;
-
+        const { allowInvalidPills, parseItems, validator } = this.props;
         let pills = parseItems ? parseItems(inputValue) : parseCSV(inputValue);
 
         if (!pills) {
             return [];
         }
 
-        if (validator) {
+        if (!allowInvalidPills) {
             pills = pills.filter(pill => validator(pill));
         }
 
-        // Keep the data format consistent with DatalistItem
         return pills.map(pill => ({
-            text: pill,
+            displayText: pill,
```

Comment:
Is this change necessary? Seems like it adds a lot of weight to the PR.

Reply:
I want to leave it to keep the data structure consistent with other drop downs. Not sure if I will get that opportunity again anytime soon. It also helps BFFs to send displayText and we not having to convert back/forth and also re-use of flow types.